client.go

```diff
@@ -933,6 +933,24 @@ func (cl *Client) sendInitialMessages(conn *connection, torrent *Torrent) {
 	}
 }
 
+func (cl *Client) sendInitialPEX(conn *connection, t *Torrent) {
+	peerPexExtendedId, ok := conn.PeerExtensionIDs[pp.ExtensionNamePex]
+	if !ok {
+		// peer did not advertise support for the PEX extension
+		conn.logger.Printf("no PEX support -  not sending initial")
+		return
+	}
+	pexMsg := t.pexInitial()
+	if pexMsg == nil {
+		// not enough peers to share — e.g. len(t.conns < 50)
```

Comment:
This doesn't match the check in `pexInitial` anymore?

Reply:
Well, it's oversimplification, and not exactly precise. The spec says if it's less that 25, we've got to add a bunch of last known good peers from the log, 25 max, for "good" defined as fully hand-shaken connections, for each of ip4 and ip6. But if there is not enough context, the initial can be delayed. Hereby the nil is the way to tell the caller it's not a good time.